p2p/README.md

```diff
@@ -106,9 +121,48 @@ ConcreteType --* OtherType : Aggregatation
 ConcreteType ..*  "(cardinality)" OtherType : Indirect (via interface)
 ```
 
+#### Interface Realization
+
+> Realization is a specialized abstraction relationship between two sets of model elements, one representing a specification (the supplier) and the other represents an implementation of the latter (the client).
+
+> Realization can be used to model stepwise refinement, optimizations, transformations, templates, model synthesis, framework composition, etc.
+
+_(see: [UML Realization](https://www.uml-diagrams.org/realization.html))_
+
+#### Direct Usage
+
+> Dependency is a directed relationship which is used to show that some UML element or a set of elements requires, needs or depends on other model elements for specification or implementation. Because of this, dependency is called a supplier - client relationship, where supplier provides something to the client, and thus the client is in some sense incomplete while semantically or structurally dependent on the supplier element(s). Modification of the supplier may impact the client elements.
+
+> Usage is a dependency in which one named element (client) requires another named element (supplier) for its full definition or implementation.
+
+_(see: [UML Dependency](https://www.uml-diagrams.org/dependency.html))_
+
+#### Composition
+
+> A "strong" form of aggregation
+
+> If a composite (whole) is deleted, all of its composite parts are "normally" deleted with it.
+
+_(see: [UML Shared composition](https://www.uml-diagrams.org/composition.html))_
+
+#### Aggregation
+
+> A "weak" form of composition
+
+> Shared part could be included in several composites, and if some or all of the composites are deleted, shared part may still exist.
+
+_(see: [UML Shared aggregation](https://www.uml-diagrams.org/aggregation.html))_
+
+#### Cardinality
+
```

Comment:
```suggestion

_tl;dr An M:N relationship defines how M instances of one type work with N instances of another type._

```

Reply:
I've only ever seen cardinality expressed in terms of "<client> has <type> association with <cardinality> number of <supplier(s)>".

I think `M:N` is an unconventional way to think about it. For example, in both composition or aggregation 
 it doesn't make sense to consider a ratio where `M` is anything other than `1` as in both association types, only one end can be the aggregation (as per the UML spec). 

I'm more than happy to take a pass at a tl;dr for this one that you can criticize. :sweat_smile:

Reply:
Here's what I went with, wdyt?:

---

#### Cardinality

_TL;DR indicates a number, or range of instances associated (i.e. supplier(s))_

Cardinality indicates the number or range of simultaneous instances of supplier that are associated with the client. Applicable to multiple association types. Can be expressed arbitrarily (e.g. wildcards, variable, equation, etc.)

Reply:
I like the tl;dr let's do it!